### ISSUE TITLE
Add DNS over HTTPS [Eny, Hara, Revrain]

### ISSUE THREAD

**Issue by RobbWatershed** [avluis/Hentoid#582]
Providers
- Cloudflare
- Quad9
- ~~Yandex (77.88.8.8)~~ <-- not a DNS over HTTP provider ("vanilla" DNS)
- ~~Linode (172.104.49.100)~~ <-- not a DNS over HTTP provider ("vanilla" DNS)

**Comment by RobbWatershed** [comment]
For the record, DNS over HTTPs won't work well with certain webpages through the Hentoid browser.

TL;DR : To be certain everything works fine, you'll have to continue using an external DNS app.

***

The core issue comes from the fact that you can't just set a new DNS to a Webview - you basically have to override all HTTP requests using `WebViewClient.shouldInterceptRequest` and run them manually with your custom DNS.

However, this sole solution has its limitations :

1/ **Pages that use plain HTTP POST** such as login pages, as Android can't override POST requests yet (see https://issuetracker.google.com/issues/36918490)

2/ **Pages that use [Range requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests)**, as Android auto-adds a `Content-Length:0` HTTP header to all responses that go through `WebViewClient.shouldInterceptRequest`, even if the returned response already has `Content-Length` set.

See https://issuetracker.google.com/issues/193071452 and https://bugs.chromium.org/p/chromium/issues/detail?id=1161877#c13 for a thorough technical explanation.

NB : These two cases hopefully both work fine when handled by the _default_ webview engine, without `WebViewClient.shouldInterceptRequest` overrides.
